File: Backend/app/schemas/tenant.py
"""Tenant schemas."""

from pydantic import BaseModel, Field
from uuid import UUID
from datetime import datetime
from typing import Optional

from app.schemas.link import LinkResponse
from app.schemas.theme import ThemeResponse


class TenantResponse(BaseModel):
    id: UUID
    slug: str
    name: str
    bio: Optional[str] = None
    avatar_url: Optional[str] = None
    social_links: Optional[dict] = None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class TenantUpdateRequest(BaseModel):
    name: Optional[str] = Field(None, max_length=255)
    bio: Optional[str] = Field(None, max_length=500)
    avatar_url: Optional[str] = Field(None, max_length=500)
    social_links: Optional[dict] = None


class PublicTenantInfo(BaseModel):
    slug: str
    name: str
    bio: Optional[str] = None
    avatar_url: Optional[str] = None
    social_links: Optional[dict] = None

    model_config = {"from_attributes": True}


class PublicProfileResponse(BaseModel):
    tenant: PublicTenantInfo
    links: list[LinkResponse] = []
    theme: Optional[ThemeResponse] = None

    model_config = {"from_attributes": True}

File: Frontend/src/types/tenant.ts
/* Tenant-related types. */

export interface Tenant {
  id: string;
  slug: string;
  name: string;
  bio: string | null;
  avatar_url: string | null;
  social_links: Record<string, string> | null;
  created_at: string;
  updated_at: string;
}

export interface TenantUpdateRequest {
  name?: string;
  bio?: string;
  avatar_url?: string;
  social_links?: Record<string, string>;
}

export interface PublicTenantInfo {
  slug: string;
  name: string;
  bio: string | null;
  avatar_url: string | null;
  social_links: Record<string, string> | null;
}

export interface PublicProfileResponse {
  tenant: PublicTenantInfo;
  links: import("./link").Link[];
  theme: import("./theme").ThemeConfig | null;
}
